src/main/java/com/cburch/logisim/fpga/gui/HdlColorRenderer.java

```diff
@@ -37,52 +40,89 @@ public HdlColorRenderer() {
   }
 
   @Override
-  public Component getTableCellRendererComponent(
-      JTable table, Object Info, boolean isSelected, boolean hasFocus, int row, int column) {
-    /* we have a difference between the first row and the rest */
-    if (row == 0) {
-      String value = (String) Info;
-      boolean passive = value.equals(NO_SUPPORT_STRING);
-      Color newColor = (passive) ? Color.red : Color.green;
-      if (value.equals(UNKNOWN_STRING)) newColor = table.getGridColor();
-      if (column == 1) setBackground(newColor);
-      setForeground(Color.black);
-      if (column == 0) setText(value);
-      else {
-        if (value.equals(NO_SUPPORT_STRING)) setText(S.get("FPGANotSupported"));
-        else if (value.equals(SUPPORT_STRING)) setText(S.get("FPGASupported"));
-        else setText(S.get("FPGAUnknown"));
-      }
+  public Component getTableCellRendererComponent(JTable table, Object infoObj, boolean isSelected,
+                                                 boolean hasFocus, int rowIdx, int columnIdx) {
+    // Some sanity checks to avoid NPE
+    final var value = (infoObj != null) ? (String) infoObj : "???";
+
+    // Set default cell colors
+    setBackground(javax.swing.UIManager.getColor("Table.background"));
+    setForeground(javax.swing.UIManager.getColor("Table.foreground"));
+
+    // First row is always a FPGS support status.
+    if (rowIdx == 0) {
+      return renderFpgaSupportRow(columnIdx, value);
+    }
+
+    // Render all the rest
+    if (value.equals(REQUIRED_FIELD_STRING)) {
+      // This field (mostly lables) is required to be set for FGPA support to work.
+      // But this is not an error if user do not really care FPGA at this moment, so let's color differently.
```

Comment:
Fix wording of this sentence: "But this is not an error if the user does not really care about FPGA at this moment, so let's color it differently."

Reply:
Done.